src/cargo/core/compiler/build_context/target_info.rs

```diff
@@ -253,16 +253,29 @@ impl TargetInfo {
                 res
             };
 
-            let cfg = lines
-                .map(|line| Ok(Cfg::from_str(line)?))
-                .filter(TargetInfo::not_user_specific_cfg)
-                .collect::<CargoResult<Vec<_>>>()
-                .with_context(|| {
-                    format!(
-                        "failed to parse the cfg from `rustc --print=cfg`, got:\n{}",
-                        output
-                    )
-                })?;
+            let cfg = {
+                let mut res = Vec::new();
+                loop {
+                    match lines.next() {
```

Comment:
I think
```rust
while let Some(line) = lines.next() {
}
if res.is_empty() {
}
```
would be clearer in intent and reduce rightward shift

Reply:
In the wider context of how this is parsing, I guess the current structure makes sense.